lib/diaspora/federation/entities.rb

```diff
@@ -200,37 +177,29 @@ def self.reshare(reshare)
         )
       end
 
-      def self.retraction(target)
+      def self.retraction(retraction)
+        case retraction.data[:target_type]
+        when "Contact"
```

Comment:
Also, if you're going to change anything in the retraction implemenation, perhaps the retraction/contact related changes could be extracted to a different pull request. If I'm not mistaken, these changes don't depend of the federation gem update and could work both with the 0.1 and 0.2 versions of diaspora_federation. So we can process these changes out of the federation gem update flow. (correct me if I'm wrong)